File: 347-TopKFrequentElements/347-TopKFrequentElements.py
# Last updated: 10/3/2025, 12:08:45 PM
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        dct={}
        for i in nums:
            if i not in dct.keys():
                dct[i]=1
            else:
                dct[i]+=1
        vst={}
        for i in dct.keys():
            vst[i]=False
        mx=[]
        for i in range(k):
            maxi=float('-inf')
            ind=None
            for k,v in dct.items():
                if v>=maxi and vst[k]==False:
                    maxi=v
                    ind=k
            
            mx.append(ind)
            vst[ind]=True
        return mx
                    

File: 347-TopKFrequentElements/347-TopKFrequentElements.cpp
// Last updated: 4/4/2026, 12:41:44 AM
class Solution {
  
public:
static bool cmp(pair<int,int>a,pair<int,int>b ){
    return b.second< a.second;
}
    vector<int> topKFrequent(vector<int>& nums, int k) {
        vector<pair<int,int>>arr;
      
    vector<int>arr2;
    int n;
    n=nums.size();
    bool visit[n];
memset(visit,false,sizeof(visit));
    for (int i=0;i<n-1;i++){
        int cnt=1;
        if(visit[i]==false){
            visit[i]=true;

        for (int j=i+1;j<n;j++){
            if (nums[i]==nums[j]){
                cnt++;
                visit[j]=true;
            }
        }
        arr.push_back({nums[i],cnt});
    }
    }
    
    vector <int>fnl;
    if(visit[nums.size()-1]==false){
        arr.push_back({nums[nums.size()-1],1});
        sort(arr.begin(),arr.end(),cmp);
        for (int i=0;i<k;i++){
            fnl.push_back(arr[i].first);
        }
        return fnl;
    }
    else{
        sort(arr.begin(),arr.end(),cmp);
        for (int i=0;i<k;i++){fnl.push_back(arr[i].first);
        }
        return fnl;
    }
    
    }
};